Add column sort and bulk action system to Shadcn AutoTable

MillanWangGadget opened the pull request:
- **UPDATE**
  - Added column sorting based on clicking column headers. Has corresponding indicators for sort direction
  - Added record selection and bulk action execution system that matches the modal vs direct callback system used in Polaris

airhorns reviewed (commented):
🎩 

- can you hide the sort chevrons until the header is hovered like polaris does? and maybe make em a bit smaller? Theyre just kinda loud if all shown at once

![CleanShot 2025-02-03 at 11 27 49@2x](https://github.com/user-attachments/assets/dedb2564-f241-4b1f-b517-23536e3a8d34)

- the delete button didn't work when i clicked it in the storybook -- is that expected?
 - there's a `huh` that snuck in somewhere lol
 
![CleanShot 2025-02-03 at 11 29 37@2x](https://github.com/user-attachments/assets/4acdea13-8a76-424f-ada4-a401f60412cd)

 - the buttons for the bulk actions seem like they are in a weird spot, want to just copy what Polaris does for organizing them / laying them out?

MillanWangGadget commented on packages/react/spec/auto/shadcn-defaults/components/Checkbox.tsx:
```diff
@@ -17,7 +17,7 @@ const Checkbox = React.forwardRef<
   >
     <CheckboxPrimitive.Indicator className={cn("flex items-center justify-center text-current")}>
-      <Check className="h-4 w-4" />
+      {props.checked === "indeterminate" ? <Minus className="h-4 w-4" /> : <Check className="h-4 w-4" />}
```
This is an update that I added to go off script from the code snip that shadcn provides. They don't have a build in  way to show a `some but not all` selected state.

If a user has the default shadcn code snip instead, they will the checked state look the same as the intermediate `some but not all` selected state. 

If this is OK or not depends on how we expect users to implement a Shadcn auto comps setup. I think that telling users to use the default shadcn components without modification is not great because they don't include pretty basic UI stuff like hover color changes, and it'd be a lot of work for the user to get something that is on the same visual level as Polaris. I think it'd be better to give users a collection of UI component files that allow us to give them a better starting point, and that would allow us to put in something like this to the default files

MillanWangGadget commented:
> * can you hide the sort chevrons until the header is hovered like polaris does? and maybe make em a bit smaller? Theyre just kinda loud if all shown at once
> 

 Updated such that they are smaller and that the unsorted indicators only appear on hover
 
 
> * the delete button didn't work when i clicked it in the storybook -- is that expected?
> * there's a `huh` that snuck in somewhere lol

I have a feeling these points were from an old commit that I pushed before asking for a review. Now they should be good and the bulk actions properly run the callbacks or make API requests to Gadget actions 


> * the buttons for the bulk actions seem like they are in a weird spot, want to just copy what Polaris does for organizing them / laying them out?

Updated such that they follow the Polaris implementation. When records are selected, they replace the column headers with the promoted actions as buttons, and other actions within the a dropdown menu

MillanWangGadget commented on packages/react/cypress/component/auto/table/AutoTableBulkActions.cy.tsx:
```diff
@@ -1,6 +1,6 @@
+import { ActionErrorMessage, ActionSuccessMessage } from "../../../../src/auto/hooks/useTableBulkActions.js";
 import { PolarisAutoTable } from "../../../../src/auto/polaris/PolarisAutoTable.js";
 import { api } from "../../../support/api.js";
 import { PolarisWrapper } from "../../../support/auto.js";
```
Updated to use `describeForEachAutoAdapter` and to run tests on the Shadcn side

airhorns commented on packages/react/cypress/component/auto/table/AutoTableBulkActions.cy.tsx:
```diff
@@ -79,23 +92,23 @@ describe("AutoTable - Bulk actions", () => {
         ]}
       />,
-      PolarisWrapper
+      wrapper
```
❤️
